### PR TITLE
Install redux

### PR DESCRIPTION
**Pull request by ianmah**
If you don't know react, this might not make much sense. If you know react, this might not make much sense. 

I was using query param to determine which event to show in the app. With Redux we can store that data in the store instead of checking against the query param.

Now, the param is only used when the app is refreshed or linked to. In that case, we find the query param, and then set up the store based on that param. If no param, display home page.

Sorry for the big PR, I should have split it up so the code is easier to understand.

I (or Jacques) will teach everyone how redux works soon

If you're doing a real CR it'll be helpful to look at it commit by commit

### CODE REVIEW

File: src/components/Nav.js [hunk 1]
```diff
@@ -12,18 +12,23 @@ import {
...
 import { ChevronLeft, Menu } from '@material-ui/icons';
+import { setEvent } from "../actions/indexActions";
+import { connect } from "react-redux";
 
-export default function Nav(props) {
+function Nav(props) {
 
   const events = props.events
 
+  // const setEvent = props.setEvent
```
Comment (jacqueschen1) [inline src/components/Nav.js]: What's this comment for?

File: src/actions/indexActions.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export function setPage(page) {
+  return {
+    type: 'SET_PAGE',
```
Comment (jacqueschen1) [inline src/actions/indexActions.js]: Should we keep a separate CONSTANTS file like in the app? It lets us more easily configure constants in the future if there's ones that need renaming

Comment (ianmah) [reply src/actions/indexActions.js]: In what case would having a variable called `FAILED` help us over a string called `LOGIN_FAILED`?
I supposed it helps the IDE?

File: src/components/Nav.js (hunk 1 above)
Comment (ianmah) [reply src/components/Nav.js]: ill get rid of it

File: src/actions/indexActions.js (hunk 2 above)
Comment (jacqueschen1) [reply src/actions/indexActions.js]: In the app? I think we could put constants together so it's like `LOGIN.FAILED` and `LOGIN.SUCCESS` or something, I think that was the original point of it

**Pull request merged**